Task-RelayedData.md

```diff
@@ -0,0 +1,104 @@
+# SaltyRTC Relayed Data Task
+
+This task uses the end-to-end encrypted WebSocket connection set up by
+the SaltyRTC protocol to send user defined messages.
+
+# Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+# Terminology
+
+All terms from the [SaltyRTC protocol
+specification](./Protocol.md#terminology) are valid in this document.
+
+# Task Protocol Name
+
+The following protocol name SHALL be used for task negotiation:
+
+`v0.relayed-data.tasks.saltyrtc.org`
+
+# Task Data
+
+This task does not currently make use of the *data* field of the 'auth'
+messages described in the [SaltyRTC protocol
+specification](./Protocol.md#auth-message). The task data sent by the
+peers should be `Nil`.
+
+# Message Structure
+
+The same message structure as defined in the [SaltyRTC protocol
+specification](./Protocol.md#message-structure) SHALL be used.
+
+# Client-to-Client Messages
+
+Once the task has taken over, user data MAY be sent using 'data'
+messages.
+
+If one of the sides wants to terminate the connection, the 'close'
+Message SHALL be sent as described in the [SaltyRTC protocol
+specification](./Protocol.md#close-message).
+
+Other message types (including the
+['application' message](./Protocol.md#application-message))
+SHALL NOT be used in the Relayed Data task.
+
+## Message States (Beyond 'auth')
+
+          +--+
+          |  v
+        +-+----+    +-------+
+    -+->+ data +--->+ close |
+     |  +------+    +-------+
+     |                  ^
+     +------------------+
+
+## 'data' messages
+
+Once the task has taken over, the user application of a client MAY
+trigger sending this message.
+
+This message MAY contain any MessagePack value type. If the user does
+not wish to encode messages with MessagePack, they can be transmitted
+using MessagePack binary or string value types.
+
+A task who sends a 'data' message SHALL set the *p* field to whatever
+data the user application provided. The name is an abbreviation that
+stands for "payload" and has been chosen to reduce the message overhead.
```

Comment:
I foretell that people will laugh at us for doing this while keeping the field name *type* and it's value *data* where we could have taken a number or whatever... and I will forward them to you. :grin:

Reply:
Feel free :P